### ISSUE TITLE
running GTDB on V2.0[question]

### ISSUE THREAD

**Issue by kusandeep** [bactopia/bactopia#262]
Hi @rpetit3,

I am wondering if we can run GTDB on bactopia 2.0, if so could you provide me an examples..?
Alternatively, I was thinking to have older version 1.7.X and run GTDB on that. If so any suggestion how can I create a separate conda env for bactopia 1.7..

conda create -n bactopia -c conda-forge -c bioconda bactopia-1.7...?

Thanks,
Sandeep

**Comment by rpetit3** [comment]
Hi @kusandeep !

How soon do you need this? I think I have most things set up for it to be in v2, so I can get it added back.

**Comment by kusandeep** [comment]
I would say the earlier the better. But you are already doing a great job and I don't want to put more pressure..

**Comment by rpetit3** [comment]
Good news! I think I have a working subworkflow for GTDB. 

Want me create a temporary conda install for you to test it out?

**Comment by rpetit3** [comment]
I might need you on this one, the system I have ran out of memory.

```
mamba create -n test-bactopia-gtdb -c rpetit3 -c conda-forge -c bioconda bactopia
conda activate test-bactopia-gtdb
bactopia --wf gtdb --help
```

**Comment by kusandeep** [comment]
seems I am having the same issue and it taking ages.
In   1-1/2 hr run it still at 
`(/dataset/SK_PMI/itmp/conda-envs/test-bactopia-gtdb) kumars@grabfopp03:/dataset/SK_PMI/itmp$ bactopia -wf  gtdb --bactopia /bifo/itmp/WenL/bactopia_out/run1/ --gtdb /dataset/datacache/scratch/bactopia/gtdbtk/ --outdir /bifo/itmp/SK_PMI/bactopia_out/wenl/ --cpus 16
2021-12-20 11:53:53:root:INFO - Found /dataset/SK_PMI/itmp/conda-envs/test-bactopia-gtdb/share/bactopia-2.0.x/conda/linux/annotate_genome.yml (1 of 7), begin build to /dataset/SK_PMI/itmp/conda-envs/test-bactopia-gtdb/share/bactopia-2.0.x/conda/envs/annotate_genome-2.0.x
2021-12-20 12:58:16:root:INFO - Found /dataset/SK_PMI/itmp/conda-envs/test-bactopia-gtdb/share/bactopia-2.0.x/conda/linux/assemble_genome.yml (2 of 7), begin build to /dataset/SK_PMI/itmp/conda-envs/test-bactopia-gtdb/share/bactopia-2.0.x/conda/envs/assemble_genome-2.0.x`

**Comment by rpetit3** [comment]
Should be `--wf gtdb` (two `--`)

**Comment by kusandeep** [comment]
oppsss..

new error now..
`WARN: Found unexpected parameters:
* --cpus: 16
- Ignore this warning: params.schema_ignore_params = "cpus"

Unknown method invocation `contains` on Integer type -- Did you mean?
  toString
`

**Comment by kusandeep** [comment]
update.

`bactopia --wf  gtdb --bactopia /bifo/itmp/WenL/bactopia_out/run1/ --gtdb /dataset/datacache/scratch/bactopia/gtdbtk/ --outdir /bifo/itmp/SK_PMI/bactopia_out/wenl/ --max_cpus 16
N E X T F L O W  ~  version 21.10.0
Launching `/dataset/SK_PMI/itmp/conda-envs/test-bactopia-gtdb/share/bactopia-2.0.x/main.nf` [sleepy_northcutt] - revision: 81a0ab6755
Unknown method invocation `contains` on Integer type -- Did you mean?
  toString

 -- Check script '/dataset/SK_PMI/itmp/conda-envs/test-bactopia-gtdb/share/bactopia-2.0.x/./workflows/bactopia-tools.nf' at line: 19 or see '.nextflow.log' file for more details`

**Comment by rpetit3** [comment]
Does it work fine without `--max_cpus` ?

**Comment by kusandeep** [comment]
naa I tried that too. same error.

**Comment by rpetit3** [comment]
Fixed it. https://github.com/bactopia/bactopia/commit/fc2f93b306f355fe905c10371a06e4e9908bad92

I'm rebuilding on my personal channel

**Comment by rpetit3** [comment]
Ok rebuilt, might need to force update it. Otherwise You can edit this file 
```
/dataset/SK_PMI/itmp/conda-envs/test-bactopia-gtdb/share/bactopia-2.0.x/modules/nf-core/modules/gtdbtk/setupdb/params.json
```

To delete line 17
```
"default": 10,
```

**Comment by kusandeep** [comment]
Ok Shall I delete 
"default": 10,  ? in line 17 or how do I update the whole env?

**Comment by rpetit3** [comment]
Deleting the line might be easier, but I think for update you would

```
conda update -c rpetit3 bactopia
```

**Comment by kusandeep** [comment]
It seems to be running now but from the last 15 min it trying to create env!!

`[-        ] process > BACTOPIATOOLS:GTDB:CLASSIFY               -
[-        ] process > BACTOPIATOOLS:CUSTOM_DUMPSOFTWAREVERSIONS -
Creating env using mamba: bioconda::gtdbtk=1.7.0 [cache /dataset/SK_PMI/itmp/conda-envs/test-bactopia-gtdb/share/bactopia-2.0.x/conda/envs/env-a1f743fa86e8334b3d87cd1168031a85]`

I'll keep you posted.

**Comment by rpetit3** [comment]
Can you by chance use docker or Singularity?

**Comment by kusandeep** [comment]
Let me check with @afmcc

**Comment by kusandeep** [comment]
> It seems to be running now but from the last 15 min it trying to create env!!
> 
> `[- ] process > BACTOPIATOOLS:GTDB:CLASSIFY - [- ] process > BACTOPIATOOLS:CUSTOM_DUMPSOFTWAREVERSIONS - Creating env using mamba: bioconda::gtdbtk=1.7.0 [cache /dataset/SK_PMI/itmp/conda-envs/test-bactopia-gtdb/share/bactopia-2.0.x/conda/envs/env-a1f743fa86e8334b3d87cd1168031a85]`
> 
> I'll keep you posted.

Definitely an issue here as it seems to be stuck on this step.

**Comment by kusandeep** [comment]
> 

and it's failed now
`   Launch Dir       : /bifo/itmp/SK_PMI

[-        ] process > BACTOPIATOOLS:GTDB:CLASSIFY               -
[-        ] process > BACTOPIATOOLS:GTDB:CLASSIFY               -
[-        ] process > BACTOPIATOOLS:CUSTOM_DUMPSOFTWAREVERSIONS -
Error executing process > 'BACTOPIATOOLS:GTDB:CLASSIFY (C6007)'

Caused by:
  Failed to create Conda environment
  command: mamba create -c conda-forge -c bioconda --mkdir --yes --quiet --prefix /dataset/SK_PMI/itmp/conda-envs/test-bactopia-gtdb/share/bactopia-2.0.x/conda/envs/env-a1f743fa86e8334b3d87cd1168031a85 bioconda::gtdbtk=1.7.0
  status : 143
  message:
`

**Comment by kusandeep** [comment]
I do not have mamba installed yet as I am using conda instead,.. Not sure if that's a part of the problem..?

**Comment by rpetit3** [comment]
Ok I've updated the bactopia build on `-c rpetit3` to use conda as the default. Let's see if that does it

```
conda create -n test-bactopia-gtdb -c rpetit3 -c conda-forge -c bioconda bactopia
conda activate test-bactopia-gtdb
bactopia --wf gtdb --help
```

**Comment by rpetit3** [comment]
Hi @kusandeep 

GTDB is now available in v2.0.1 (https://github.com/bactopia/bactopia/releases/tag/v2.0.1), going to close for now. Please feel free to reopen

Robert
